### PR TITLE
Fix inconsistent rendering of block mathematical expressions

### LINKED ISSUE #28735: Inconsistent rendering of block mathematical expressions
### Description

According to https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions

```
$$\left( \sum_{k=1}^n a_k b_k \right)^2 \leq \left( \sum_{k=1}^n a_k^2 \right) \left( \sum_{k=1}^n b_k^2 \right)$$

```math
\left( \sum_{k=1}^n a_k b_k \right)^2 \leq \left( \sum_{k=1}^n a_k^2 \right) \left( \sum_{k=1}^n b_k^2 \right)
`
```
Should both render the same
$$\left( \sum_{k=1}^n a_k b_k \right)^2 \leq \left( \sum_{k=1}^n a_k^2 \right) \left( \sum_{k=1}^n b_k^2 \right)$$

```math
\left( \sum_{k=1}^n a_k b_k \right)^2 \leq \left( \sum_{k=1}^n a_k^2 \right) \left( \sum_{k=1}^n b_k^2 \right)
```

However, in Gitea the backtick block is rendered left-aligned and more like an inline expression (see screenshot).

### Gitea Version

1.21.3

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

gitea-math

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

As a Systemd service.

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by yp05327**
Fix #28735

GitHub render `\```math\``` ` as a block now.
Add `display` class will render it as a block.

After:
image
image

### CODE REVIEW

**Comment by silverwind** [comment]
Where is this `display` class from? I would recommend using tailwind, e.g. `tw-block` for `display: block`.

**Comment by yp05327** [comment]
https://github.com/go-gitea/gitea/blob/0c273f12e0b5d6dc2b80e2e6b51b2deaf15608b1/web_src/js/markup/math.js#L32-L46
It is used for select the display mode in js. It doesn't look like a class for css.

**Comment by silverwind** [comment]
Ah I see, the JS is then turning it into `katex-display` in the output which does the block rendering.

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
Refactor for the future: Turn `display` class into `data-is-display` for clarity.

File: modules/markup/markdown/markdown.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func SpecializedMarkdown() goldmark.Markdown {
 							}
 
 							// include language-x class as part of commonmark spec
-							_, err = w.WriteString(`<code class="chroma language-` + string(language) + `">`)
+							_, err = w.WriteString(`<code class="chroma language-` + string(language) + ` display">`)
```
Comment (wxiaoguang) [inline modules/markup/markdown/markdown.go:108]:
```suggestion
							// the "display" class is used by "js/markup/math.js" to render the code element as a block
							_, err = w.WriteString(`<code class="chroma language-` + string(language) + ` display">`)
```

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**